### DISCUSSION TITLE
Is there a way mocking GraphQL requests without operationName?

### DISCUSSION THREAD

**Discussion by bennyk8y** [mswjs/msw#1632] [Q&A] [answered]
We are using https://genql.dev/ and it doesnt send operationName by default. it means that this code won't work 
```js
import { setupServer } from 'msw/node';  
setupServer(graphql.query('users', (req, res, ctx) => res());   
```

because it cannot detect that the query is "users" as in the request itself there's no operationName property. Is there any other way to do it?

**Comment by kettanaito** [comment, marked as answer]
Hi, @bennyk8y. I believe I've answered a very similar question on Discord.

You can use the [graphql.operation()](https://mswjs.io/docs/api/graphql/operation) API to capture any outgoing GraphQL operations, even if they don't have operation names. Naturally, since there's no identifiers to distinguish between them on the library level, you have to implement some sort of predicate logic to know how to respond to each operation. 

The easiest approach here is to resolve any outgoing operation against the actual GraphQL schema you're using and a mocked data set. Here's an [example](https://github.com/mswjs/msw/blob/da92465b312b54a7a7e71168b1bdcb62daeef3b1/test/node/graphql-api/compatibility.node.test.ts#L19) of doing that.
